Assets/Scripts/LifeManager.cs

```diff
@@ -35,6 +48,28 @@ public void Die()
 			mPhotonView.RPC("OnDeath", PhotonTargets.MasterClient);
 		}
 
+        public void InflictDamage(int damageAmount)
+        {
+            Assert.IsTrue(damageAmount >= 0);
+            mCurHealth = Mathf.Max(0, mCurHealth - damageAmount);
+
+            mPhotonView.RPC("UpdateLocalHealthBar", PhotonTargets.All, mCurHealth);
+
+            if (mCurHealth == 0)
+            {
+                Die();
+            }
+        }
+
+        [PunRPC]
+        public void UpdateLocalHealthBar(int health)
+        {
+            if (mIsLocalPlayer)
```

Comment:
in line with the previous comment, instead of guarding against mIsLocalPlayer, we should just only send the RPC to the local player